flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/remote/RemoteStorageScanner.java

```diff
@@ -144,7 +157,9 @@ public void watchSegment(
 
     /** Close the executor. */
     public void close() {
-        scannerExecutor.shutdownNow();
+        synchronized (scannerExecutor) {
+            scannerExecutor.shutdownNow();
```

Comment:
I'd suggestion waiting for all pending task become finished when close this scanner.

Reply:
Done. I have added `scannerExecutor .awaitTermination`.